### PR TITLE
Rails JS Bundling: Fix minor grammatical errors

### PR DESCRIPTION
**Pull request by abarnes1**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [ ] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Fixes several minor grammar issues within the Rails JS Bundling lesson.

**2. This PR:**

- Corrects your/you're and its/it's issues
- Uppercase Ruby and Rails when not part of a command
- Fixes capitalization at the start of a few sentences
- Fixed a few sentences to remove fragments and run-ons
- Misc: missing spaces, comma in the wrong spot

**3. Additional Information:**

Some terms are not very consistent throughout the lesson.  I am happy to make additional changes to address these with some guidance as to how they should be referred to:
- jsbundling/JS Bundling/JSBundling
- esbuild/Esbuild/ESBuild
- javascript/Javascript/JavaScript

### CODE REVIEW

**Comment by ChargrilledChook** [comment]
Hey @abarnes1, thanks for this contribution. If you're happy to do a consistency pass on the last three things you mentioned, that would be great. My preference would be:
* JavaScript (official name)
* esbuild (stylised as such in their own docs)

The last one I'm not sure, but it probably doesn't matter as long as it's consistent. JavaScript Bundling  is probably best (maybe a bit wordy?) but I don't think it's a huge deal. I'm happy for your to use your judgement.

[Their repo for reference](https://github.com/rails/jsbundling-rails)

**Comment by abarnes1** [comment]
I went with JS bundling when referring to the concept of bundling JavaScript and the gem name jsbundling-rails when directly addressing the Rails implementation of the bundling concept.  If that's wrong it's at least consistently wrong.

**Review by ChargrilledChook — APPROVED** [review]
Thanks @abarnes1, nice work :rocket:
